apps/emqx_connector_aggregator/test/emqx_connector_aggreg_json_lines_tests.erl

```diff
@@ -0,0 +1,78 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024-2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_connector_aggreg_json_lines_tests).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-export([decode/1]).
+
+%%------------------------------------------------------------------------------
+%% Helper fns
+%%------------------------------------------------------------------------------
+
+new_opts() ->
+    #{}.
+
+fill_close(Records, JSONL) ->
+    {Output, JSONLFinal} = emqx_connector_aggreg_json_lines:fill(Records, JSONL),
+    Trailer = emqx_connector_aggreg_json_lines:close(JSONLFinal),
+    iolist_to_binary([Output, Trailer]).
+
+decode(Binary) ->
+    do_decode(Binary, []).
+
+do_decode(Binary, Acc) ->
+    case jiffy:decode(Binary, [return_maps, return_trailer]) of
+        {has_trailer, Term, Rest} ->
+            do_decode(Rest, [Term | Acc]);
+        Term ->
+            lists:reverse([Term | Acc])
+    end.
+
+rountrip(Records, JSONL) ->
+    Binary = fill_close(Records, JSONL),
+    decode(Binary).
+
+%%------------------------------------------------------------------------------
+%% Test cases
+%%------------------------------------------------------------------------------
+
+roundtrip_test() ->
+    JSONL = emqx_connector_aggreg_json_lines:new(new_opts()),
+    ?assertEqual(
+        [
+            #{<<"A">> => 1.2345, <<"B">> => "string", <<"Ç"/utf8>> => +0.0},
+            #{<<"A">> => 1 / 3, <<"B">> => "[]", <<"Ç"/utf8>> => -0.0},
+            #{<<"A">> => 111111, <<"B">> => "🫠", <<"Ç"/utf8>> => 0.0},
+            #{<<"A">> => 111.111, <<"B">> => "\"quoted\"", <<"Ç"/utf8>> => "line\r\nbreak"},
```

Comment:
...Asking because this has a chance of becoming `"Ç": [108,105,110,101,13,10,98,114,101,97,107]` in the resulting document, but I'm probably missing something.

Reply:
Yes, apparently our JSON encoder does convert Erlang strings to arrays of ints.

I assume that who generates the data should use binary instead of erlang strings if they wish to avoid this.

Reply:
Sure, and I think this very test does verify that it is indeed encoded as an array of ints, given that the decoded output is an erlang string instead of binary.  Indeed:

```elixir
iex(emqx@127.0.0.1)3> :emqx_utils_json.encode(%{a: ~c"line\r\nbreak"})
"{\"a\":[108,105,110,101,13,10,98,114,101,97,107]}"
```

And with binary value:

```elixir
iex(emqx@127.0.0.1)1> :emqx_utils_json.encode(%{a: "line\r\nbreak"})
"{\"a\":\"line\\r\\nbreak\"}"
```

I'll add an extra case for the latter.

Reply:
Thanks, good to know!